test/kmedoids.jl

```diff
@@ -30,14 +30,14 @@ R = kmedoids(costs, k)
 # group 3: [6, 8, 9], values: [21, 20, 22]
 #
 
-X = reshape(@compat(map(Float64, [1, 6, 2, 3, 7, 21, 8, 20, 22])), 1, 9)
+X = reshape(map(Float64, [1, 6, 2, 3, 7, 21, 8, 20, 22]), 1, 9)
 costs = pairwise(SqEuclidean(), X)
 
 R = kmedoids!(costs, [1, 2, 6])
 @test isa(R, KmedoidsResult)
 @test R.medoids == [3, 5, 6]
 @test R.assignments == [1, 2, 1, 1, 2, 3, 2, 3, 3]
 @test R.counts == [3, 3, 3]
-@test_approx_eq R.acosts [1, 1, 0, 1, 0, 0, 1, 1, 1]
-@test_approx_eq R.totalcost 6.0
+@test all(isapprox.(R.acosts, [1, 1, 0, 1, 0, 0, 1, 1, 1]))
```

Comment:
No, `@compat(Union{...})` is for 0.3 support; the 0.3-style `Union(...)` is deprecated in 0.4.

I'm not opposed to dropping 0.4 support, I was mostly just wondering whether it was actually necessary to allow 0.6 to work.